src/rest/handlers/rest_queries.erl

```diff
@@ -101,28 +101,95 @@ execute(Fun, Args, FieldName) ->
+        {ok, Members} -> {ok, #{<<"reaches">> => sets:is_element(From, Members), <<"duration">> => 0}}
+    end.
 
 %% @todo
 -spec reaches_indexed(binary(), binary()) -> {ok, boolean()} | {error, any()}.
 reaches_indexed(_From, _To) ->
     {ok, false}.
 
 %% @todo
--spec effective_permissions_naive(binary(), binary()) -> {ok, binary()} | {error, any()}.
-effective_permissions_naive(_From, _To) ->
-    {ok, <<"">>}.
+-spec effective_permissions_naive(binary(), binary()) -> {ok, map()} | {error, any()}.
+effective_permissions_naive(From, To) ->
+    MyZoneId = gmm_utils:zone_id(),
+    case gmm_utils:owner_of(From) of
+        MyZoneId -> {ok, #{<<"effectivePermissions">> => do_effective_permissions_naive(From, To), <<"duration">> => 0}};
+        Other -> zone_client:effective_permissions(naive, Other, From, To)
+    end.
+
+do_effective_permissions_naive(From, To) ->
+    JoinPermissions = fun
+        (<<"no-route">>, Permissions) -> Permissions;
+        (Permissions, <<"no-route">>) -> Permissions;
+        (A, B) -> gmm_utils:permissions_or(A,B)
+    end,
+    case graph:list_children(From) of
+        {ok, Children} ->
+            case lists:member(To, Children) of
+                true -> {ok, graph:get_edge(From, To)};
+                false ->
+                    lists:foldl(fun
+                        (_, {error, Error}) -> {error, Error};
+                        (Child, {ok, Permissions}) ->
+                            case effective_permissions_naive(Child, To) of
+                                {ok, #{<<"effectivePermissions">> := ChildPermissions}} -> {ok, JoinPermissions(ChildPermissions, Permissions)};
+                                {error, Error} -> {error, Error}
+                            end
+                        end,
+                        {ok, <<"no-route">>},
+                        Children)
+            end;
+        {error, Error} -> {error, Error}
+    end.
 
 %% @todo
 -spec effective_permissions_indexed(binary(), binary()) -> {ok, binary()} | {error, any()}.
 effective_permissions_indexed(_From, _To) ->
     {ok, <<"">>}.
 
 %% @todo
--spec members_naive(binary()) -> {ok, list(binary())} | {error, any()}.
-members_naive(_Of) ->
-    {ok, []}.
+-spec members_naive(binary()) -> {ok, map()} | {error, any()}.
+members_naive(Of) ->
+    case do_members_naive(Of) of
+        {ok, Set} -> {ok, #{<<"members">> => sets:to_list(Set), <<"duration">> => 0}};
+        {error, Error} -> {error, Error}
+    end.
+
+-spec do_members_naive(binary()) -> {ok, sets:set(binary())} | {error, any()}.
+do_members_naive(Of) ->
+    MyZoneId = gmm_utils:zone_id(),
+    case gmm_utils:owner_of(Of) of
+        MyZoneId -> do_do_members_naive(Of);
+        Other ->
+            case zone_client:members(naive, Other, Of) of
+                {ok, #{<<"members">> := Members}} -> sets:from_list(Members);
+                {error, Error} -> {error, Error}
+            end
+    end.
+
+do_do_members_naive(Of) ->
```

Comment:
Ta nazwa XDD zmień to na coś czytelniejszego

Reply:
+1 XD